### PR TITLE
Package dynamic iOS framework

### PR DESCRIPTION
**Pull request by jpsim**
CI still needs to be updated. This addresses #1747 /cc @segiddins @tgoyne 

Installation instructions are in https://github.com/realm/realm.io/pull/452.

### CODE REVIEW

File: scripts/strip-frameworks.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
Comment (segiddins) [inline scripts/strip-frameworks.sh]: should probably include our license header?

**Comment by segiddins** [comment]
Is it intentional that this script will strip more than just the Realm framework?

**Comment by jpsim** [comment]
Yes, it is intentional that this script will strip more than just the Realm framework.

File: scripts/strip-frameworks.sh (hunk 1 above)
Comment (jpsim) [inline scripts/strip-frameworks.sh]: I don't know, should it? Our other scripts don't include it, so is that an omission?

File: scripts/strip-frameworks.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  if ! [[ "$(file "$file")" == *"dynamically linked shared library"* ]]; then
+    continue
+  fi
+  # Get architectures for current file
+  archs="$(lipo -info "${file}" | rev | cut -d ':' -f1 | rev)"
+  stripped=""
+  for arch in $archs; do
+    if ! [[ "${VALID_ARCHS}" == *"$arch"* ]]; then
+      # Strip non-valid architectures in-place
+      lipo -remove "$arch" -output "$file" "$file" || exit_code=1
```
Comment (segiddins) [inline scripts/strip-frameworks.sh]: would it be better to just fail-fast here?

Comment (jpsim) [inline scripts/strip-frameworks.sh]: I'm not sure what you mean.

Comment (mrackwitz) [inline scripts/strip-frameworks.sh]:
```diff
-…|| exit_code=1
+…|| exit 1
```

Comment (jpsim) [inline scripts/strip-frameworks.sh]: Ah, I see. No, I'd prefer not because this way, we will continue to strip other frameworks, which is the intention.

Comment (segiddins) [inline scripts/strip-frameworks.sh]: But since we'll `exit 1` at the end, the build will fail anyways?

Comment (jpsim) [inline scripts/strip-frameworks.sh]: Alright, good point. Early failure it is.

**Comment by segiddins** [comment]
This looks fine to me.

**Comment by tgoyne** [comment]
:+1:

**Pull request merged**